### PR TITLE
[RAPTOR-5369] improve error output

### PR DESCRIPTION
**Pull request by yakov-g**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
I was able to reproduce the error mentioned in the ticket by failing server to bind the port, thus py4j client has nothing to connect to.
So a couple of improvements are:
1. If py4j server fails to bind to a port (in PredictorEntryPoint.java), explicitly exit the program.
2. On the python side: if py4j server failed to start, print a full output.
3. I'd like to release DRUM v 1.5.6.post1 and reinstall java env, which will pull this DRUM, so we could see a proper log then it fails. (Though i think it will be a binding error, if so - i have some following steps in mind)

### CODE REVIEW

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java [hunk 1]
```diff
@@ -86,15 +86,18 @@ private static EntryPointConfig parseArgs(String[] args) throws Exception {
...
         try {
             EntryPointConfig config = parseArgs(args);
+            portNumber = config.portNumber;
             PredictorEntryPoint entryPoint = new PredictorEntryPoint(config.className, config.predictorName);
             // TODO: use a specific port, note multiple such gateways might be running.
             GatewayServer gatewayServer = new GatewayServer(entryPoint, config.portNumber);
             gatewayServer.start();
         } catch (Exception e) {
-            System.out.println("Got exception in ComponentEntryPoint: " + e.getMessage());
-            throw e;
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java:97]: Looks like GatewayServer starts some thread, thus throwing exception doesn't finish the program.
Call exit explicitly.

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java [hunk 2]
```diff
@@ -86,15 +86,18 @@ private static EntryPointConfig parseArgs(String[] args) throws Exception {
...
+            System.out.println(String.format("PredictorEntryPoint failed to start py4j GatewayServer on port: %d", portNumber));
```
Comment (zohar-mizrahi) [inline custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java:98]: Don't you want to print the message from the exception? (.e.g `e.getMessage()`)

**Review by zohar-mizrahi — COMMENTED** [review]
How easy it to add a test case to check the new functionality?

File: custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java (hunk 2 above)
Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java:98]: In the example I play with, trying to bind to port 80.
```
null
PredictorEntryPoint failed to start py4j GatewayServer on port: 80
py4j.Py4JNetworkException
	at py4j.GatewayServer.startSocket(GatewayServer.java:788)
	at py4j.GatewayServer.start(GatewayServer.java:763)
	at py4j.GatewayServer.start(GatewayServer.java:746)
	at com.datarobot.drum.PredictorEntryPoint.main(PredictorEntryPoint.java:96)
Caused by: java.net.BindException: Permission denied (Bind failed)
	at java.base/java.net.PlainSocketImpl.socketBind(Native Method)
	at java.base/java.net.AbstractPlainSocketImpl.bind(AbstractPlainSocketImpl.java:436)
	at java.base/java.net.ServerSocket.bind(ServerSocket.java:395)
	at py4j.GatewayServer.startSocket(GatewayServer.java:786)
	... 3 more

```

In this exception message passed is `null`, that's how it is implemented:

https://github.com/bartdag/py4j/blob/366de7a527df122f4c5b50a5aa10fb69cd47a089/py4j-java/src/main/java/py4j/GatewayServer.java#L788

https://github.com/bartdag/py4j/blob/366de7a527df122f4c5b50a5aa10fb69cd47a089/py4j-java/src/main/java/py4j/Py4JNetworkException.java#L79

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java:98]: Or do you mean to print the message in the addition to the stacktrace?

Comment (yakov-g) [reply custom_model_runner/datarobot_drum/drum/language_predictors/java_predictor/src/main/java/com/datarobot/drum/PredictorEntryPoint.java:98]: added message printing

**Comment by yakov-g** [comment]
> How easy it to add a test case to check the new functionality?

added

**Review by zohar-mizrahi — APPROVED** [review]

**Review by jmbannon — APPROVED** [review]

**Pull request merged**